pages/content/posts/2022-q3-report/index.md

```diff
@@ -0,0 +1,186 @@
+
+* Modern JavaScript is now easier to read and write.
+* Most projects use modern JavaScript or transpilers.
+* Some patterns used in the 2000s are now considered anti-patterns to avoid.
+
+**In Q3 2022, we agreed which modern JS features to use on MDN, then applied them to most of our static examples.**
+
+We proceeded in three phases.
+
+1. First, we drove an open discussion to define JS features that are in common use now. The discussion was vivid and civil and was noticed in the JS community. The newsletter “JS Weekly” had a mention about it in one of its episodes, and 19 people participated in the discussion with a total of 81 messages; much more than usual MDN discussions. In that discussion, a consensus was reached about which JS features to use. Examples are, among other things:
+    - using `for…of` or `.forEach` instead of classic `for(;;)` structure,
+    - `let` and `const` instead of `var`
+    - template literals,
+    - arrow functions
+
+2. In the second phase, we applied what has been consensus to most static examples of MDN. _We created a community of contributors to help us_. With the help of XYZ contributors and XYZ PRs, we updated most static examples of MDN. We limited ourselves to the most common features, and this changed the way MDN examples are perceived: from old-style coding examples, they are now modern-style examples.
+3. Finally, the third phase consisted of updating MDN's JS coding guidelines. We also worked by consensus between all maintainers of MDN and created a single PR with 298 comments. This allowed us to communicate the new style to most maintainers but also to be able to refer to agreed conventions when divergences occur.
+
+MDN is so large that there is still a lot to do after this project. For now, we focused on static examples on MDN pages. MDN also has interactive examples and a corpus of external examples (js-examples, css-examples, dom-examples, …); these examples should also be updated to the new guidelines. We hope to do this in the future. Still, as this is a significant piece of work, we would like to group this with a few other modernization activities that are needed: we would like to lint these examples with ESlint (to detect errors), avoid outdated APIs (like using the Fetch API instead of XHR), as well as fix any accessibility and semantic issues. We think it is a good idea to triage these structural issues, drive discussions about best practices and then apply the changes to all MDN pages.
+
+To sum up, we renovated all static examples of MDN. They use modern JS and can be more easily copied and pasted into real projects. They also teach better practices to developers. Working with the community on this project has been gratifying, and as it helped to finalize the whole project in a quarter when otherwise these mass-changes on thousands of MDN pages can take much longer.
```

Comment:
```suggestion
To sum up, we renovated all static examples on MDN. They now use modern JS and can be more easily copied and pasted into real projects. They also teach better practices to developers. Working with the community on this project has been gratifying, and it helped us finish the whole project in a quarter when otherwise these mass-changes on thousands of MDN pages can take much longer.
```